index.ts

```diff
@@ -0,0 +1,104 @@
+const NativeClient = require('bindings')('hawk_tracer_client').HawkTracerClient;   // loads build/Release/hawk_tracer_client.node
+
+export interface HawkTracerClientOptions {
+    source: string;
+    map_files?: string;
+}
+
+// HT_Event
+interface BareEvent {
+    klass_id: number;
+    timestamp: number;
+    id: number;
+}
+
+// HT_EventKlassInfoEvent
+interface EventKlassInfoEvent extends BareEvent {
+    info_klass_id: number;
+    event_klass_name: string;
+    field_count: number;
+}
+
+function isEventKlassInfoEvent(event: BareEvent): event is EventKlassInfoEvent {
+    const e = event as EventKlassInfoEvent;
+    return e.info_klass_id !== undefined && e.event_klass_name !== undefined;
```

Comment:
I don't think this comparison is strong enough. Somebody might define a custom event with those two fields, which is not eventklassinfo event. We should be rather using `klass_id` for comparison to make sure this is the right class.

Reply:
I tried to avoid depending on global state, since it's a special type of functions called [type guards](https://www.typescriptlang.org/docs/handbook/advanced-types.html#type-guards-and-type-assertions).
While it may limit the usability a bit, I'll change it to check for `klass_id` and make it normal function.